src/BranchAndBound.jl

```diff
@@ -0,0 +1,168 @@
+struct BnBNode
+    LD::AbstractLagrangeDual
+    lb::Float64
+end
+
+mutable struct BnBTree
+    incumbent::Vector{BnBNode}     # list of nodes
+    feasible_prob::Dict{Int,JuMP.Model}
+    z_ub::Float64
+    z_lb::Float64
+    best_solution
+
+    function BnBTree(LD::AbstractLagrangeDual)
+        BT = new()
+
+        root = BnBNode(LD,-Inf)
+        BT.nodes = [root]
+        BT.feasible_prob = get_problem_copy(LD)
+        BT.z_ub = Inf
+        BT.z_lb = -Inf
+        best_solution = nothing
+
+        return BT
+    end
+end
+
+
+
+function get_problem_copy(LD::AbstractLagrangeDual)::Dict{Int,JuMP.Model}
+    feasible_prob = Dict()
+    for s in parallel.getpartition()
+        m = copy(LD.block_model.model[s])
```

Comment:
Copying a `JuMP.Model` may be very expensive. @hideakiv can you check the timing of this part?

Reply:
Prefer `copy_model`:
https://github.com/jump-dev/JuMP.jl/blob/7ef4be423397c82a59e210bb9c0fb67671951ce1/src/copy.jl#L71-L106

Does this work otherwise? Are you just using the `m[:name]` lookups to match variables between the two copies? This could lead to hard-to-debug issues.

Reply:
@odow I appreciate your input. We are currently trying to avoid copying the problems overall.